sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregators/Aggregator.java

```diff
@@ -14,20 +14,21 @@
  * limitations under the License.
  */
 
-package io.opentelemetry.sdk.metrics;
+package io.opentelemetry.sdk.metrics.aggregators;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /** Aggregator represents the interface for all the available aggregations. */
 @ThreadSafe
-interface Aggregator<T extends Aggregator<?>> {
```

Comment:
we had added this to avoid the instanceOf checks in the merge methods. Why remove it here?

Reply:
With generics it becomes very hard to encapsulate this class in another class. For example to use this in AbstractBoundInstrument I have to use generics there as well (maybe I am too new in generics in Java).

Reply:
Also I commented in the description why I changed this.

Reply:
I'm not sure I follow the reasoning yet. Hard to see in isolation in this PR, rather that in the context of where you want to use it. I'd be ok if you wanted to get things working, then I could go back and see if I could clean up the overall structure, so we don't have to be doing instanceof all the time.